internal/metrics/metrics.go

```diff
@@ -0,0 +1,92 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// When adding metric names, see https://prometheus.io/docs/practices/naming/#metric-names
+const (
+	createdEdgeDevicesQuery                      = "k4e_operator_created_edge_devices"
+	registeredEdgeDevicesQuery                   = "k4e_operator_edge_devices_were_registered"
+	failedToCompleteRegistrationEdgeDevicesQuery = "k4e_operator_edge_devices_failed_to_register"
+	createdEdgeDeploymentsQuery                  = "k4e_operator_created_edge_deployment"
+	createdOBCQuery                              = "k4e_operator_created_obc"
+)
+
+var (
+	createdEdgeDevices = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: createdEdgeDevicesQuery,
+			Help: "Number of created EdgeDevices",
+		},
+	)
+	registeredEdgeDevices = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: registeredEdgeDevicesQuery,
+			Help: "Number of registered EdgeDevices",
+		},
+	)
+	failedToCompleteRegistrationEdgeDevices = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: failedToCompleteRegistrationEdgeDevicesQuery,
+			Help: "Number of failed registration EdgeDevices",
+		},
+	)
+	createdEdgeDeployments = prometheus.NewGauge(
```

Comment:
pls replace with https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Counter
it will replace the need to re-calculate the numbers.

Reply:
Counter is used only for incrementing, the number of edgedeployments can be increased and decreased, isn't the `Gauge` the proper type, here?

Reply:
Yes, you are correct. I missed the fact `Counter` cannot be decreased.
My intention was to avoid https://github.com/jakub-dzon/k4e-operator/pull/46/files#diff-0e245eaea02ac9a12c64a8b1e4ec99514a84e7066ca56aac23f9f0591c71fab4R69 for listing and summerizing the amount of edge-devices in each loop.

Hence instead of using the `createdEdgeDevices.Set(float64(value))` to use `createdEdgeDevices.Inc())` when registering new device (or when creating if it is implemented in the edge-device controller) and to use `createdEdgeDevices.Decc())` when edge device is de-registered.

Reply:
Specifically for the scale testing I'd like to capture the number of successful registration vs failures.
For that purpose counter can be a good option (the numbers are expected only to grow).

Reply:
After discussing about it we decided to use Counter for registration (succeeded and failures), and OBCs creation 
The creation of edge devices and edge deployment not relevant for the scaling tests